#! /bin/sh /usr/share/dpatch/dpatch-run
##
## DP: This allows to configure the recorde buffer size and increases the
## DP: default buffer size from 5 MB to 100 MB.
## DP:
## DP: When encountering buffer overflows, the recordbuffer size should be
## DP: increased.

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' vdr1.7+~/config.c vdr1.7+/config.c
--- vdr1.7+~/config.c	2011-10-25 22:04:26.000000000 +0200
+++ vdr1.7+/config.c	2011-10-25 22:06:47.000000000 +0200
@@ -399,6 +399,7 @@
   InitialVolume = -1;
   ChannelsWrap = 0;
   EmergencyExit = 1;
+  RecorderBufSize = 100;
 }
 
 cSetup& cSetup::operator= (const cSetup &s)
@@ -591,6 +592,7 @@
   else if (!strcasecmp(Name, "InitialVolume"))       InitialVolume      = atoi(Value);
   else if (!strcasecmp(Name, "ChannelsWrap"))        ChannelsWrap       = atoi(Value);
   else if (!strcasecmp(Name, "EmergencyExit"))       EmergencyExit      = atoi(Value);
+  else if (!strcasecmp(Name, "RecorderBufSize"))     RecorderBufSize    = atoi(Value);
   else
      return false;
   return true;
@@ -687,6 +689,7 @@
   Store("InitialVolume",      InitialVolume);
   Store("ChannelsWrap",       ChannelsWrap);
   Store("EmergencyExit",      EmergencyExit);
+  Store("RecorderBufSize",    RecorderBufSize);
 
   Sort();
 
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' vdr1.7+~/config.h vdr1.7+/config.h
--- vdr1.7+~/config.h	2011-10-25 22:04:26.000000000 +0200
+++ vdr1.7+/config.h	2011-10-25 22:06:47.000000000 +0200
@@ -290,6 +290,7 @@
   int InitialVolume;
   int ChannelsWrap;
   int EmergencyExit;
+  int RecorderBufSize;
   int __EndData__;
   cString InitialChannel;
   cSetup(void);
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' vdr1.7+~/recorder.c vdr1.7+/recorder.c
--- vdr1.7+~/recorder.c	2011-10-25 22:04:26.000000000 +0200
+++ vdr1.7+/recorder.c	2011-10-25 22:08:48.000000000 +0200
@@ -10,7 +10,7 @@
 #include "recorder.h"
 #include "shutdown.h"
 
-#define RECORDERBUFSIZE  (MEGABYTE(5) / TS_SIZE * TS_SIZE) // multiple of TS_SIZE
+#define RECORDERBUFSIZE  (MEGABYTE(Setup.RecorderBufSize) / TS_SIZE * TS_SIZE) // multiple of TS_SIZE
 
 // The maximum time we wait before assuming that a recorded video data stream
 // is broken:
@@ -32,6 +32,8 @@
   SpinUpDisk(FileName);
 
   ringBuffer = new cRingBufferLinear(RECORDERBUFSIZE, MIN_TS_PACKETS_FOR_FRAME_DETECTOR * TS_SIZE, true, "Recorder");
+  dsyslog("RECORDERBUFSIZE: %d\n", RECORDERBUFSIZE);
+
   ringBuffer->SetTimeouts(0, 100);
 
   int Pid = Channel->Vpid();
